.github/workflows/publish.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish to NPM
+on:
+  release:
+    types: [created]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        id: install-pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+          run_install: false
+
+      - name: Setup Node
+        id: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install dependencies and build 🔧
+        id: install-deps
+        run: pnpm run build
+
+      - name: Publish package on NPM 📦
```

Comment:
MAYBE: do you want to run this for every release, or only ones that follow a certain pattern (such as starting with `v`, as in `v1.0.0`)?

For reference, I did the latter with MWA, to allow creating, e.g., test releases that don't do the publish step: https://github.com/solana-mobile/mobile-wallet-adapter/blob/main/.github/workflows/android.yml#L91